### DISCUSSION TITLE
Use Avalonia with VSIX

### DISCUSSION THREAD

**Discussion by loremipsumowski** [AvaloniaUI/Avalonia#11016] [Q&A]
Hello,

I am developing VSIX extension using for Sql Server Management Studio. Could I use there in any way Avalonia components? I tried a few approaches but every was without success. I created project in Visual Studio of "VSIX Project" type and I have UserControl (xaml) added by default. I would like to use there Avalonia component and I tried by adding namespace like below:

```
<UserControl
             xmlns:av="https://github.com/avaloniaui"
             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
    <av:TreeDataGrid></av:TreeDataGrid>
</UserControl>
```

but I have error: "The name "TreeDataGrid" does not exists in the namespace "https://github.com/avaloniaui".". Maybe should I add there reference to assembly, or do something else. I am not expert unfortunately but I am quite certain that is possible to use Avalonia with vsix extension.

**Comment by workgroupengineering** [comment]
Visual Studio uses WPF. You can embed Avalonia in WPF. See [this](https://github.com/AvaloniaUI/Avalonia/tree/master/samples/interop/WindowsInteropTest)
